header-rewrite-filter/header_processor.cc

```diff
@@ -190,13 +190,15 @@ namespace HeaderRewriteFilter {
         return absl::OkStatus();
     }
 
-    absl::Status SetBoolProcessor::executeOperation() {
+    std::pair<absl::Status, bool> SetBoolProcessor::executeOperation(bool negate) {
+        bool result;
         try {
-            result_ = matcher_(source_);
+            result = matcher_(source_);
         } catch (std::exception& e) {
-            return absl::InvalidArgumentError("failed to perform boolean match");
+            return std::make_pair(absl::InvalidArgumentError("failed to perform boolean match"), false);
         }
-        return absl::OkStatus();
+        result = negate ? !result : result;
```

Comment:
Nit: this could be the tiniest bit safer by using a new variable, instead of reassigning `result`, and making both `const`